src/simple_cache_client.rs

```diff
@@ -1524,36 +1524,82 @@ impl SimpleCacheClient {
         cache_name: &str,
         list_name: impl IntoBytes,
         range: impl RangeBounds<u32>,
+    ) -> MomentoResult<()> {
+        self.list_erase_many(cache_name, list_name, vec![range])
+            .await
+    }
+
+    /// Erase multiple ranges of elements from a list.
+    ///
+    /// *NOTE*: This is preview functionality and requires that you contact
+    /// Momento Support to enable these APIs for your cache.
+    ///
+    /// # Arguments
+    ///
+    /// * `cache_name` - the name of the cache in which to look for the list.
+    /// * `list_name` - the name of the list from which to remove elements.
+    /// * `ranges` - ranges of indices to erase from the list.
+    ///
+    /// # Example
+    /// ```
+    /// # fn main() -> momento_test_util::DoctestResult {
+    /// # momento_test_util::doctest(|cache_name, auth_token| async move {
+    /// use std::time::Duration;
+    /// use momento::{CollectionTtl, SimpleCacheClientBuilder};
+    ///
+    /// let ttl = CollectionTtl::default();
+    /// let mut momento = SimpleCacheClientBuilder::new(auth_token, Duration::from_secs(30))?
+    ///     .build();
+    ///
+    /// momento.list_set(&cache_name, "list", ["a", "b", "c", "d", "e", "f"], ttl).await?;
+    /// momento.list_erase_many(&cache_name, "list", vec![1..=2, 4..=4]).await?;
+    ///
+    /// let entry = momento.list_fetch(&cache_name, "list").await?.unwrap();
+    /// let values: Vec<_> = entry.value().iter().map(|v| &v[..]).collect();
+    /// let expected: Vec<&[u8]> = vec![b"a", b"d", b"f"];
+    /// assert_eq!(values, expected);
+    /// # Ok(())
+    /// # })
+    /// # }
+    ///
+    pub async fn list_erase_many<R: RangeBounds<u32>>(
+        &mut self,
+        cache_name: &str,
+        list_name: impl IntoBytes,
+        ranges: Vec<R>,
```

Comment:
do you need both vec and a bounded R here or could you use `&[impl Into<RangeBounds<u32>>]`?

Reply:
You do need the bounded `R` since `RangeBounds` is a trait but there's no reason for it to have to be a `Vec`. I'll change it to an `impl IntoIterator<Item = R>`.